proposals/0000-a11y-roles-states-improvements.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Improvements to accessibilityRole and accessibilityStates
+author:
+- Marc Mulcahy
+date: 2018-10-31
+---
+
+## Summary
+
+Some brilliant work was begun to add cross platform support for accessibility roles and states to React Native. We propose to add some additional roles and states, as well as extend the implementation on Android and Fire OS.
+
+## Basic example
+
+With the current set of roles, it is not possible to describe a checkbox. These changes would enable this scenario, among other things:
+
+    <View
+      accessibilityRole="checkbox"
+      accessibilityStates={['unchecked']}>
+      <Text>Show Password</Text>
+    </View>
+
+## Motivation
+
+When developers build components in JavaScript, they need the ability to express the role and current state(s) of the component to assistive technologies running on the platform.
+
+## Detailed design
+
+### New Roles
+
+This proposal adds the following roles:
+
+- alert
+- checkbox
+- combobox
+- editabletext
+- menu
+- menubar
+- menuitem
+- progressbar
+- radiobutton
+- radiogroup
+- scrollbar
+- spinbutton
+- tab
+- tablist
+- timer
+- toolbar
+
+### New States
+
+This proposal adds the following states:
```

Comment:
`disabled`

Reply:
Looks like disabled is already supported?